xarray/core/resample_cftime.py

```diff
@@ -360,3 +455,19 @@ def exact_cftime_datetime_difference(a, b):
     seconds = int(round(seconds.total_seconds()))
     microseconds = b.microsecond - a.microsecond
     return datetime.timedelta(seconds=seconds, microseconds=microseconds)
+
+
+def _convert_offset_to_timedelta(
+    offset: datetime.timedelta | str | BaseCFTimeOffset,
+) -> datetime.timedelta:
+    if isinstance(offset, datetime.timedelta):
+        return offset
+    elif isinstance(offset, (str, *CFTIME_TICKS)):
+        return to_offset(offset).as_timedelta()
+    else:
+        raise ValueError
+
+
+def _ceil_via_cftimeindex(date, freq):
```

Comment:
You can define the type at the top of the module in a `if typing.TYPE_CHECKING` block (Maybe an `else: the type: typing.Any?)
We do that in the types module e.g.

Reply:
Thanks -- I added a `CFTimeDatetime` type in `types.py`, since it seems like it might be useful elsewhere.  I also did my best to add types for the functions in `resample_cftime.py`.

Reply:
I'll see if I can iterate locally on the mypy errors raised in CI.